src/server/tokens-db.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+let mongodb = require('mongodb');
+let MongoClient = mongodb.MongoClient;
+let protocol = process.env.DB_PROTOCOL;
+let host = process.env.DB_HOST;
+let port = process.env.DB_PORT;
+let name = process.env.DB_NAME;
+let url = protocol+'://' + host+':' + port + '/' + name;
+function createToken(userId, userAgent) {
+  let tokenDescriptor = {};
+  let days = 7;
+  let expiresAt = new Date().getTime() + days * 24 * 60 * 60 * 1000;
+  let userAgentToString;
+  tokenDescriptor.userId = userId;
+  tokenDescriptor.userAgent = userAgent;
+  tokenDescriptor.expiresAt = expiresAt;
+  if (tokenDescriptor.userAgent.includes('Windows')) {
+    userAgentToString = 'qwerasdf';
+  } else if (tokenDescriptor.userAgent.includes('Mac')) {
+    userAgentToString = 'asdfzxcv';
+  } else if (tokenDescriptor.userAgent.includes('iPhone')) {
+    userAgentToString = 'zxcvasdf';
+  } else if (tokenDescriptor.userAgent.includes('Android')) {
+    userAgentToString = 'asdfqwer';
+  } else {
+    userAgentToString = 'zxcvzxcv';
+  }
+  tokenDescriptor.token = tokenDescriptor.userId + userAgentToString;
+  // confirm if the token already exited
+  searchTokenDescriptor(tokenDescriptor, insertTokenDescriptor);
+  return {
+    token: tokenDescriptor.token,
+    expiresAt: tokenDescriptor.expiresAt,
+  };
+}
+// delete Token from db
+function deleteToken(userId) {
```

Comment:
delete token works by token not userId.
when we will call the logout, we will pass a token, and then we just need to delete token regarding to the authorization header